File: COB-ts/src/pmi.ts
import { round2 } from './money.js';
import { validatePmiPaymentInputs } from './validate.js';

/** Monthly PMI payment against a given balance (typically the original loan amount —
 *  PMI doesn't vary month-to-month like a running-balance figure would). */
export function calculatePmiPayment(loanBalance: number, annualRatePercent: number): number {
  validatePmiPaymentInputs(loanBalance, annualRatePercent);
  return round2((loanBalance * annualRatePercent) / 100 / 12);
}

File: COB-py/cob_calculator/pmi.py
from .money import round2
from .validate import validate_pmi_payment_inputs


def calculate_pmi_payment(loan_balance: float, annual_rate_percent: float) -> float:
    """Monthly PMI payment against a given balance (typically the original loan
    amount -- PMI doesn't vary month-to-month like a running-balance figure would)."""
    validate_pmi_payment_inputs(loan_balance, annual_rate_percent)
    return round2((loan_balance * annual_rate_percent) / 100 / 12)
